### PR TITLE
Implement remote.WriteIndex

### PR DESCRIPTION
**Pull request by jonjohnsonjr**
Fixes https://github.com/google/go-containerregistry/issues/119

### CODE REVIEW

**Comment by jonjohnsonjr** [comment]
Rebased!

File: pkg/v1/remote/write.go [hunk 1]
```diff
@@ -369,4 +398,61 @@ func scopesForUploadingImage(ref name.Reference, layers []v1.Layer) []string {
 	return scopes
 }
 
-// TODO(mattmoor): WriteIndex
+// WriteIndex pushes the provided Imageindex to the specified image reference.
```
Comment (ImJasonH) [inline pkg/v1/remote/write.go]: ImageIndex

File: pkg/v1/remote/write_test.go [hunk 2]
```diff
@@ -152,18 +160,17 @@ func setupWriterWithServer(server *httptest.Server, repo string, img v1.Image) (
 
 	return &writer{
 		ref:    tag,
-		img:    img,
 		client: http.DefaultClient,
 	}, server, nil
 }
 
-func TestCheckExistingFound(t *testing.T) {
+func TestCheckExistingBlobFound(t *testing.T) {
```
Comment (ImJasonH) [inline pkg/v1/remote/write_test.go]: Can this be a table driven test with the next couple tests?

Comment (jonjohnsonjr) [reply pkg/v1/remote/write_test.go]: Good thinking

File: pkg/v1/remote/write.go (hunk 1 above)
Comment (jonjohnsonjr) [reply pkg/v1/remote/write.go]: Nice catch

**Pull request merged**